sdk/Sdk.Generators/Extensions/ISymbolExtensions.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Generators
+{
+    internal static class ISymbolExtensions
+    {
+        /// <summary>
+        /// Walks the symbol tree to generate the fully qualified name of a type symbol.
+        /// Ex input: A Symbol for "Task" token
+        /// Output: "System.Threading.Tasks.Task"
+        /// </summary>
+        /// <param name="typeSymbol"></param>
+        /// <returns></returns>
+        internal static string GetFullName(this ITypeSymbol typeSymbol)
+        {
+            var symbol = typeSymbol as ISymbol;
+
+            if (symbol == null || IsRootNamespace(symbol))
+            {
+                return string.Empty;
+            }
+
+            var sb = new StringBuilder(symbol.MetadataName);
+
+            if (symbol is IArrayTypeSymbol arraySymbol) // arrays need to be handled differently b/c the properties used to get the full name for other symbols are null for IArrayTypeSymbols
+            {
+                sb.Append(arraySymbol.ElementType.GetFullName()); // ex: for string[], the ElementType is System.String and that is the full name returned at this step.
```

Comment:
Besides the array check this isn't my code - I just moved the file into a different folder. I also think we're removing this method b/c both @kshyju and I refactored to use SymbolEqualityComparer instead of relying on this, but these changes are in different PRs and it would be messy if I removed it right now...

If @kshyju 's refactoring gets merged soon I can rebase and then remove this method in this PR, but otherwise I think it may be best to save it for another PR.